x/tokenomics/token_logic_module/settlement_results.go

```diff
@@ -0,0 +1,162 @@
+package token_logic_module
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"cosmossdk.io/log"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// SettlementResults is a slice of SettlementResult. It implements
+// methods for convenience when working with SettlementResult objects.
+type SettlementResults []*tokenomicstypes.SettlementResult
+
+// resultOption is a function which receives a SettlementResult for modification.
+type resultOption func(*tokenomicstypes.SettlementResult)
+
+// NewSettlementResult returns a new SettlementResult with the given claim and options applied.
+func NewSettlementResult(
+	claim prooftypes.Claim,
+	opts ...resultOption,
+) *tokenomicstypes.SettlementResult {
+	result := &tokenomicstypes.SettlementResult{Claim: claim}
+	for _, opt := range opts {
+		opt(result)
+	}
+	return result
+}
+
+// GetNumComputeUnits returns the total number of claimed compute units in the results.
+func (rs SettlementResults) GetNumComputeUnits() (numComputeUnits uint64, errs error) {
+	for _, result := range rs {
+		claimNumComputeUnits, err := result.GetNumComputeUnits()
+		if err != nil {
+			errs = errors.Join(err, err)
+			continue
+		}
+		numComputeUnits += claimNumComputeUnits
+	}
+
+	return numComputeUnits, errs
+}
+
+// GetNumRelays returns the total number of relays in the combined results.
+func (rs SettlementResults) GetNumRelays() (numRelays uint64, errs error) {
+	for _, result := range rs {
+		claimNumRelays, err := result.Claim.GetNumRelays()
+		if err != nil {
```

Comment:
ULTRA NIT: Here an below: should we ever check if Claim is nill or is there no chance?

Reply:
It should not be possible. `result` is a protobuf type and its `Claim` field is non-nullable.